sources/vcdMerge/src/SignalSource.cpp

```diff
@@ -97,11 +99,28 @@ void MERGE::SignalSource::Create()
 
 void MERGE::SignalSource::SetFormat(const std::string &rFormat)
 {
-    if (rFormat != "T")
+    if (rFormat == "T")
     {
-        throw EXCEPTION::VcdException(EXCEPTION::Error::INVALID_LOG_FILE_FORMAT,
-                                      "Invalid log file format: " + rFormat);
+        m_pSignalFactory = std::make_unique<PARSER::SignalFactory>();
+        return;
     }
+
+    if (rFormat.length() > 3)
+    {
+        if ((rFormat[0] == 'U') &&
+            (rFormat[1] == '{') &&
+            (rFormat[rFormat.length() - 1] == '}'))
+        {
+            const std::string filename(rFormat.substr(2, rFormat.length() - 3));
```

Comment:
Don't use `substr`. Use one of `std::string` constructors:
```cpp
const std::string filename(rFormat, 2, rFormat.length() - 3);
```
Magic numbers btw.

Reply:
These magic numbers are just numbers of characters. It wouldn't change much if I would change '3' to 'THREE'. I will provide some enhanced comments.

Reply:
But they mean something. It's 3 for a reason, and this "reason" should be a name of const.